web-marketplace/src/pages/CreditClassDetails/CreditClassDetails.tsx

```diff
@@ -237,11 +241,17 @@ function CreditClassDetails({
         isBuyButtonDisabled={isBuyFlowDisabled && Boolean(wallet?.address)}
         isCommunityCredit={isCommunityCredit}
         onBookCallButtonClick={onBookCallButtonClick}
-        onBuyButtonClick={() =>
-          isBuyFlowDisabled
-            ? setConnectWalletModal(atom => void (atom.open = true))
-            : setIsBuyFlowStarted(true)
-        }
+        onBuyButtonClick={() => {
+          if (isBuyFlowDisabled) {
+            setConnectWalletModal(atom => void (atom.open = true));
+          } else {
+            if (isConnected) {
+              setIsBuyFlowStarted(true);
+            } else {
+              setSwitchWalletModalAtom(atom => void (atom.open = true));
```

Comment:
if I'm logged in with a web2 account, then I get the following:

![image](https://github.com/regen-network/regen-web/assets/3149358/ea9e9f6d-9611-4ed6-b3f5-1c16f45776d9)

Reply:
Yes I'm getting this too

Reply:
I made a change to always check the `activeAccount.addr` instead of the `wallet.addr` like before. The`KeplrWalletConnectModal` is now displayed properly.

Reply:
I have applied the same logic everywhere.

Reply:
Fair point, I have fixed that here: [7203217](https://github.com/regen-network/regen-web/pull/2227/commits/720321747b2a8b3e3a91bcba8c77d563fee65f22)